src/app/routes/connector-ui/contract-agreement-page/contract-agreement-cards/contract-agreement-card-mapped.service.ts

```diff
@@ -43,4 +45,25 @@ export class ContractAgreementCardMappedService {
   ): ContractAgreementCardMapped[] {
     return search(cards, searchText, (card) => card.searchTargets);
   }
+
+  setStatus(
+    maxConsumingContracts: number,
+    agreements: ContractAgreementCardMapped[],
+  ): ContractAgreementCardMapped[] {
+    return agreements.map((it, index) => ({
+      ...it,
+      isConsumingLimitsEnforced: true,
+      statusText: index < maxConsumingContracts ? 'Active' : 'Inactive',
+      statusTooltipText:
+        index >= maxConsumingContracts
+          ? maxConsumingContracts == 1
+            ? 'This connector is configured with a limit of ' +
+              maxConsumingContracts +
```

Comment:
nice job for smoothing out the edges!

Regarding code structure, isn't it easier and better DRY if one used template strings? (Better for the reviewer, too, so I don't get a knot in my head)

e.g. like this:

```ts
var msg = `This is a long long text about pears. However, you have ${numApples} apple${numApples ? 's' : ''}. Some other long text about bananas.`
```